src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,195 @@
+
+    if (trimmedTitle === '') {
+      deleteTodoItem(todo.id);
+    } else if (trimmedTitle !== todo.title) {
+      setIsLoading(true);
+
+      await delay(100);
+
+      todoService
+        .updateTodo({
+          ...todo,
+          title: trimmedTitle,
+        })
+        .then(updatedTodo => {
+          setTodos(prev => prev.map(t => (t.id === todo.id ? updatedTodo : t)));
+          setEditTitle(trimmedTitle);
+          setIsEditing(false);
+        })
+        .catch(() => {
+          setErrorMessage(Errors.UpdateError);
+
+          if (editInputRef.current) {
+            editInputRef.current.focus();
+          }
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    } else {
+      setIsEditing(false);
+    }
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      handleSave();
+    } else if (event.key === 'Escape') {
+      setIsEditing(false);
+      setEditTitle(todo.title);
+    }
+  };
+
+  return (
+    <>
+      <div
+        data-cy="Todo"
+        className={cn('todo', {
+          completed: todo.completed,
+        })}
+      >
+        <label className="todo__status-label">
+          <input
+            data-cy="TodoStatus"
+            type="checkbox"
+            className="todo__status"
+            onChange={() =>
+              updateTodo({ ...todo, completed: !todo.completed } as Todo)
+            }
+            checked={todo.completed}
+          />
+        </label>
+
+        {isEditing ? (
+          <form
+            onSubmit={e => {
+              e.preventDefault();
+            }}
+          >
+            <input
+              data-cy="TodoTitleField"
+              type="text"
+              className="todo__title-field"
+              placeholder="Empty todo will be deleted"
+              ref={editInputRef}
+              value={editTitle}
+              onChange={e => setEditTitle(e.target.value)}
+              onBlur={handleSave}
+              onKeyUp={handleKeyUp}
+            />
+          </form>
+        ) : (
+          <>
+            <span
+              data-cy="TodoTitle"
+              className="todo__title"
+              onDoubleClick={handleDoubleClick}
+            >
+              {editTitle}
+            </span>
+
+            <button
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              onClick={() => deleteTodoItem(todo.id)}
+            >
+              ×
+            </button>
+          </>
+        )}
+
+        <div
+          data-cy="TodoLoader"
+          className={cn('modal', 'overlay', {
+            'is-active':
+              todo.id === 0 || isLoading || updatingTodos.includes(todo.id),
+          })}
+        >
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div>
+      </div>
+    </>
```

Comment:
looks like a redundant fragment

https://react.dev/reference/react/Fragment#usage

Reply:
Thanks for the feedback!